File: inst/pagebreak.hs
#!/usr/bin/env runhaskell

import Text.Pandoc.JSON

pgBrkXml :: String

pgBrkXml = "<w:p><w:r><w:br w:type=\"page\"/></w:r></w:p>"

pgBrkBlock :: Block
pgBrkBlock = RawBlock (Format "openxml") pgBrkXml

insertPgBrks :: Block -> Block
insertPgBrks (RawBlock (Format "latex") "\\newpage") = pgBrkBlock
insertPgBrks blk = blk

main = toJSONFilter insertPgBrks

File: inst/pagebreak.R
#!/usr/bin/env Rscript

json_in <- file('stdin', 'r')

lat_newp <- '{"t":"RawBlock","c":["latex","\\\\newpage"]}'

doc_newp <- '{"t":"RawBlock","c":["openxml","<w:p><w:r><w:br w:type=\\"page\\"/></w:r></w:p>"]}'

ast <- paste(readLines(json_in, warn=FALSE), collapse="\n")

ast <- gsub(lat_newp, doc_newp, ast, fixed=TRUE)

write(ast, "")

